org.idempiere.webservices/WEB-INF/src/org/compiere/model/MWebServiceType.java

```diff
@@ -286,5 +286,45 @@ public X_WS_WebServiceFieldInput getFieldInput(String colName){
 		
 		return m_inputFieldMap.get(colName);
 	}
+
+	public static void insertParameters (Properties ctx, MWebServiceType ws, String trxName) {
+
+		if (ws.getWS_WebService().getValue().equals("ModelADService")) {
+			String method = ws.getWS_WebServiceMethod().getValue();
+
+			if (method.equals("getList")) {
+				addWsParameter(ctx, ws, "AD_Reference_ID", X_WS_WebService_Para.PARAMETERTYPE_Free, "", trxName);
+			}
+			else if (method.equals("runProcess")) {
+				addWsParameter(ctx, ws, "AD_Process_ID", X_WS_WebService_Para.PARAMETERTYPE_Constant, "", trxName); // can't fill it as the process is unknown
+				addWsParameter(ctx, ws, "AD_Menu_ID", X_WS_WebService_Para.PARAMETERTYPE_Constant, "0", trxName);
+				addWsParameter(ctx, ws, "AD_Record_ID", X_WS_WebService_Para.PARAMETERTYPE_Free, "", trxName);
+			}
+			else {
+				String value = "";
+				if (method.equals("createData")) value = "Create";
+				if (method.equals("deleteData")) value = "Delete";
+				if (method.equals("queryData")) value = "Read";
+				if (method.equals("readData")) value = "Read";
+				if (method.equals("updateData")) value = "Update";
+
+				addWsParameter(ctx, ws, "TableName", X_WS_WebService_Para.PARAMETERTYPE_Constant, MTable.get(ctx, ws.getAD_Table_ID()).getTableName(), trxName);
+				addWsParameter(ctx, ws, "Action", X_WS_WebService_Para.PARAMETERTYPE_Constant, value, trxName);
+				addWsParameter(ctx, ws, "RecordID", X_WS_WebService_Para.PARAMETERTYPE_Free, "", trxName);
+			}
+		}
+	}
 	
+	private static void addWsParameter(Properties ctx, MWebServiceType ws, String name, String type, String value, String trxName) {
+		
+		if (DB.getSQLValueEx(trxName, "SELECT 1 FROM WS_WebService_Para WHERE WS_WebServiceType_ID = ? AND ParameterName = ?", ws.getWS_WebServiceType_ID(), name) != 1) {
+			X_WS_WebService_Para wsp = new X_WS_WebService_Para(ctx, 0, trxName);
```

Comment:
There is a MWebServicePara - is always preferred to use the M class instead of the X_ class, using the X_ class skips the validations added in beforeSave.

Reply:
Correct, here's the reason : I've developped the code on a plugin based on org.adempiere.base (and MWebServicePara is not reachable from there)